### DISCUSSION TITLE
New pump model

### DISCUSSION THREAD

**Discussion by Rearth** [Rearth/Oritech#9] [Todos]
The current pump model is horrible. The idea was to create something like we see in real life to pump oil, but the implementation never really looked good.

**Comment by Rearth** [comment]
![2024-05-03_20 46 20](https://github.com/Rearth/Oritech/assets/10100603/7d3aeb17-93f7-4a3f-a703-f1f26cbc0a37)
This is the current pump model

**Comment by jshipley** [comment]
I like the idea of that pump. That design lends itself to a better animation than something like a centrifugal pump (or a single block "you don't have to know how it works" pump).

That said, here are the specific problems I see with that design:

* It's unclear what the big orange square on the side is (power input? fluid output? decoration?)
* The pump arm clips through the support legs when it's pumping in a very noticable way
* The pump shaft should be more vertical
* The box on the ground that the shaft moves in should be taller, so that you can't see the end of the shaft when it's at the highest position

Part of the problem is that an animated pump like this just doesn't fit well inside one block. Maybe a 2x1 or 2x2x1 machine would look better?

**Comment by jshipley** [comment]
![Screenshot 2024-08-01 182435](https://github.com/user-attachments/assets/5b66e178-36e0-4255-850f-0edb3eb7fdd6)

Here's a super rough concept I put together that uses a mechanism more like the steam engine than a pump jack. Inspired by the steam engine, I was kind of thinking of having a motor (larger box) turning a shaft which would raise and lower the shaft in the smaller box. Everything would need to be moved around, resized, decorated, textured, etc. to make it fit in with the rest of the mod.

Or here's a really rough idea for a larger pump jack:

![Screenshot 2024-08-01 190802](https://github.com/user-attachments/assets/dd2ed1f0-1bdf-40af-b58a-16024f4cf355)

What do you think, are either of these worth expanding on? Of the two, I think I still like the pump jack better. It would also fit in better and will be more fun to watch in motion.

**Comment by jshipley** [comment]
I put a little more effort into the pump jack. It could still use a bit more decoration (and maybe some fluid/energy i/o ports like the steam engine has) and a lot of texturing work, but I'm liking it so far.

I also think that the well needs to move back a bit so that it's centered in the block, so that it will line up with the pump trunk underneath it.

Thoughts?

![PumpJack](https://github.com/user-attachments/assets/67111c80-b7c0-45be-976b-907f65d8cab6)

**Reply by CreateModFan154** [reply]
looks like pumpjack

**Comment by Rearth** [comment]
Wow that already looks amazing, the moving parts feel really smooth. It is 2x1x1 in size?
One thing that I'd maybe change is the base plate. This is also an issue with the current model. If it's 2 blocks long, then usually only one is on solid ground, with the other half floating in the air. So a different base layout might reflect that better

**Reply by jshipley** [reply]
I'm not sure if it's 2x1x1 or 2x2x1, but I can try to keep it 2x1x1. I think as long as the resting state is only one block high, it's fine for the geckolib animation to go above that when the arm is moving.

I'll make some changes to the base plate to make this more into a solid block under the motor and open block over the fluid being pumped.

**Comment by Dreadp1r4te** [comment]
Looks good but to be honest it looks to small. An oil pump should be a multiblock in my opinion.

**Comment by jshipley** [comment]
There are a couple of reasons that I never finished that pump:

1. Getting the animation smooth was really tricky. I couldn't make a loop of connected joints, so I had to keep shifting a part to make it look like it was connected and no matter what I did it was still oddly jerking around.

2. I was having a hard time texturing the pump block in a way that looked good.

3. The more I played with Oritech, the more I liked the little pump that already existed.

4. My hard drive crashed and I lost the WIP pump block I had been working on.
